main.go

```diff
@@ -66,53 +72,115 @@ const (
 	webhookPort = 9443
 )
 
-func runPrometheusServer(metricsAddr string, tlsOptions common.SSPTLSOptions) error {
+// This callback executes on each client call returning a new config to be used
+// please be aware that the APIServer is using http keepalive so this is going to
+// be executed only after a while for fresh connections and not on existing ones
+func getConfigForClient(ctx context.Context, cfg *tls.Config, cache cache.Cache) (*tls.Config, error) {
+	var sspList ssp.SSPList
+	err := cache.List(ctx, &sspList)
+	if err != nil {
+		return nil, err
+	}
+
+	var tlsProfile *ocpconfigv1.TLSSecurityProfile
+
+	if len(sspList.Items) != 0 {
+		tlsProfile = sspList.Items[0].Spec.TLSSecurityProfile
+	}
+
+	if tlsProfile == nil {
+		cfg.MinVersion = crypto.DefaultTLSVersion()
+		cfg.CipherSuites = nil
+		return cfg, nil
+	}
```

Comment:
```suggestion
	if len(sspList.Items) == 0 {
		cfg.MinVersion = crypto.DefaultTLSVersion()
		cfg.CipherSuites = nil
		return cfg, nil
	}
	
	tlsProfile := sspList.Items[0].Spec.TLSSecurityProfile
```

Reply:
Do it like this?

Reply:
We still need to check if `spec.TLSSecurityProfile` is nil or not, so we would need to do it like this

```
if len(sspList.Items) == 0 || sspList.Items[0].Spec.TLSSecurityProfile == nil {
  cfg.MinVersion = crypto.DefaultTLSVersion()
  cfg.CipherSuites = nil
  return cfg, nil
}

tlsProfile := sspList.Items[0].Spec.TLSSecurityProfile
```